src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -557,6 +591,7 @@ private int addToFingerprint(
     static final class Builder {
       @Nullable private final Sequence<?> list;
       @Nullable private final NestedSet<?> nestedSet;
+      @Nullable private final Class<?> nestedSetElementType;
```

Comment:
I think might as well change the type of this builder field to `StringificationType`. That way it's non-null and you can move the `Class` -> `StringificationType` conversion to the constructor for the `NestedSet` case, so it's clear that is the only case that sets it to something other than `DEFAULT`. Then it can just be passed through to `VectorArg`.